teos/api.py

```diff
@@ -66,32 +61,37 @@ class API:
     Args:
         host (:obj:`str`): the hostname to listen on.
         port (:obj:`int`): the port of the webserver.
+        rw_lock (:obj:`RWLockWrite <readwritelock.rwlock.RWLockWrite>`): lock that must be acquired before reading or
+            writing to the watchtower's state.
         inspector (:obj:`Inspector <teos.inspector.Inspector>`): an ``Inspector`` instance to check the correctness of
             the received appointment data.
         watcher (:obj:`Watcher <teos.watcher.Watcher>`): a ``Watcher`` instance to pass the requests to.
 
     Attributes:
         logger: the logger for this component.
+        app: the Flask app of the API server.
     """
 
-    def __init__(self, host, port, inspector, watcher):
+    def __init__(self, host, port, rw_lock, inspector, watcher):
         self.logger = get_logger(component=API.__name__)
         self.host = host
         self.port = port
+        self.rw_lock = rw_lock
         self.inspector = inspector
         self.watcher = watcher
-        self.app = app
+
+        # ToDo: #5-add-async-to-api
+        self.app = Flask(__name__)
```

Comment:
The app may need to be outside the object so it can be called by `gunicorn` or `uWSGI` (not completely sure about this yet).

Leaving this here until I can figure it out.